test/support/case.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Alice.Handlers.Case do
```

Comment:
I think we should rename this `Alice.HandlerCase`

Reply:
Also I think we can use [ExUnit.CaseTemplate](https://hexdocs.pm/ex_unit/ExUnit.CaseTemplate.html) here, which would let us omit the extra `use ExUnit.Case` in tests.

Reply:
Ok, so I removed the need to use both `ExUnit.Case` and `Alice.HandlersCase`, however not through `ExUnit.CaseTemplate`. The problem I had with CaseTemplate was that I'm using the options passed to `__using__`  to build the `setup`. Since CaseTemplate takes over the `__using__` macro, and just provide you with normal methods to do setup and teardown. To add the handlers in the manner I'm doing at the moment, I need to be able to create the `setup` method during `__using__` when it's still a macro.

Unless I'm missing something?

Reply:
Ah yeah, right on, looks good